recipe/meta.yaml

```diff
@@ -27,9 +27,18 @@ source:
       - linuxboost.patch  # [linux]
 
 build:
-  number: 14
+  number: 15
   skip: true  # [win]
 
+# NOTE: Top-level environment with boost-cpp is only to separate the build for
+#       multiple boost versions into different CI jobs.
+# TODO: Needs investigation why "separate the two boost builds more strictly"
+#       would be necessary. Ref:
+#       https://github.com/conda-forge/mshr-feedstock/pull/23#issuecomment-749520383
+requirements:
+  host:
+    - boost-cpp
```

Comment:
@jan-janssen, since you weren't happy with my changes, I added this.
This is still just a workaround. I don't know if it's necessary. If it is, the cause should be investigated.
Prime andidates for causes are always:
- reused CMake caches (but `build-libdolfin.sh` seems to work in a fresh `build` dir already, so maybe not this)
- unexpected behavior from `conda-build` with `outputs`, e.g., https://github.com/conda/conda-build/issues/3993

Reply:
Thanks